src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -178,6 +179,53 @@ const DomainFragment = ({
     if (searchText.length > 0) fetchData();
   };
 
+  const SECTIONS = [
+    {
+      title: 'First',
+      content: 'Lorem ipsum...'
+    }
+  ];
+
+  const AccordionView = ({data}) => {
+    const [activeSections, setActiveSections] = React.useState([]);
+
+    const renderSectionTitle = () => {
+      return <View style={styles.content}></View>;
+    };
+
+    const renderHeader = (data, index) => {
+      return (
+        <DiscoveryTitleSeparator
+          key="user-title-separator"
+          text="Domains you follow"
+          showArrow
+          rotateArrow={activeSections?.some((actived) => actived === index)}
+        />
+      );
+    };
+
+    const renderContent = () => {
+      return (
+        <View style={styles.content}>{data?.map((item, index) => renderItem({index, item}))}</View>
+      );
+    };
+
+    const updateSections = (activeSectionsParams) => {
+      setActiveSections(activeSectionsParams);
+    };
+
+    return (
+      <Accordion
+        sections={SECTIONS}
+        activeSections={activeSections}
+        renderSectionTitle={renderSectionTitle}
+        renderHeader={renderHeader}
+        renderContent={renderContent}
+        onChange={updateSections}
+      />
+    );
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `AccordionView` component is defined inside another function. This could lead to unnecessary re-renders and performance issues as the component will be redefined every time the parent function is called. Consider moving this component definition outside of the parent function or use `useCallback` hook to memoize it.

```diff
- const AccordionView = ({data}) => {
+ const AccordionView = React.useCallback(({data}) => {
    // ... rest of the code
- };
+ }, []);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->